fendermint/vm/topdown/proof-service/src/assembler.rs

```diff
@@ -0,0 +1,293 @@
+        //   - nextConfigurationNumber: For power change tracking
+        // Events:
+        //   - NewTopDownMessage: Captures topdown messages for this subnet
+        //   - NewPowerChangeRequest: Captures validator power changes
+        let storage_specs = vec![
+            StorageProofSpec {
+                actor_id: self.gateway_actor_id,
+                // Calculate slot for subnets[subnetKey].topDownNonce in the mapping
+                slot: calculate_storage_slot(&self.subnet_id, TOPDOWN_NONCE_STORAGE_OFFSET),
+            },
+            StorageProofSpec {
+                actor_id: self.gateway_actor_id,
+                // nextConfigurationNumber is a direct storage variable at slot 20
+                // Using an empty key with the slot offset to get the direct variable
+                slot: calculate_storage_slot("", NEXT_CONFIG_NUMBER_STORAGE_SLOT),
+            },
+        ];
+
+        let event_specs = vec![
+            // Capture topdown messages for this specific subnet
+            EventProofSpec {
+                event_signature: NEW_TOPDOWN_MESSAGE_SIGNATURE.to_string(),
+                // topic_1 is the indexed subnet address
+                topic_1: self.subnet_id.clone(),
+                actor_id_filter: Some(self.gateway_actor_id),
+            },
+            // Capture ALL power change requests from the gateway
+            // These affect validator sets and need to be processed
+            EventProofSpec {
+                event_signature: NEW_POWER_CHANGE_REQUEST_SIGNATURE.to_string(),
+                // No topic_1 filter - we want all power changes
+                topic_1: String::new(),
+                actor_id_filter: Some(self.gateway_actor_id),
+            },
+        ];
+
+        tracing::debug!(
+            highest_epoch,
+            storage_specs_count = storage_specs.len(),
+            event_specs_count = event_specs.len(),
+            "Configured proof specs"
+        );
+
+        // Create LotusClient for this request (not stored due to Rc/RefCell)
+        let lotus_client = self.create_client();
+
+        // Generate proof bundle in blocking task
+        // CRITICAL: The proofs library uses Rc/RefCell internally making LotusClient and
+        // related types non-Send. We must use spawn_blocking to run the proof generation
+        // in a separate thread, then use futures::executor::block_on to bridge the
+        // async/sync worlds. This prevents blocking the main tokio runtime while
+        // handling non-Send types correctly.
+        let bundle = tokio::task::spawn_blocking(move || {
+            // Use futures::executor to run async code without blocking the parent runtime
+            futures::executor::block_on(generate_proof_bundle(
+                &lotus_client,
+                &parent_api,
+                &child_api,
+                storage_specs,
+                event_specs,
+            ))
```

Comment:
`LotusClient` passed to ` generate_proof_bundle` uses async `reqwest::Client`, which uses Tokio. Is it really safe to block on this future within `tokio::task::spawn_blocking` context?